bin/minikube-start

```diff
@@ -2,21 +2,47 @@
 set -euo pipefail
 source "$(dirname "$0")/.common"
 
+BLUE='\033[0;34m'
+RED='\033[0;31m'
+BOLD='\e[1m'
+RESET_EVERYTHING='\e[0m'
+
 ensure_program_in_path "minikube"
 check_minikube_version
 
 COMMAND="minikube"
-cores=$(nproc)
+CORES=$(nproc)
+# This is rounded to the nearest integer e.g. 32.8 gb will show up as 33. This is fine for the calculations we're going to do.
+TOTAL_MEM_GB=$(free --giga | grep Mem | sed 's/  */ /g' | cut -d ' ' -f 2)
+# By default, use 1/2 of the hosts memory for minikube
+AMOUNT_OF_MEMORY_TO_USE_GB=$((TOTAL_MEM_GB/2))
+if [ "$AMOUNT_OF_MEMORY_TO_USE_GB" -lt "9" ]; then
+  # However, if we got less memory than 9GB, use 3/4 of the total memory
+  AMOUNT_OF_MEMORY_TO_USE_GB=$(echo "$TOTAL_MEM_GB / 1.3333333333" | bc)
```

Comment:
```suggestion
  AMOUNT_OF_MEMORY_TO_USE_GB=$(echo "$TOTAL_MEM_GB * 0.75" | bc)
```

Reply:
The reason I did that was to make it output whole numers. I can, however, make it more clear.

Reply:
That makes sense, forgot about `bc`-command requiring that. Alternative option would be 
```bash
  AMOUNT_OF_MEMORY_TO_USE_GB=$(echo "($TOTAL_MEM_GB * 0.75) / 1" | bc)
```
This way there wouldn't be a need to use floating points with many decimals.